### PR TITLE
Added hardware keyboard shortcuts for iPad

### LINKED ISSUE #806: Add hardware keyboard shortcuts
Let's add keyboard shortcuts for the various editor features for devices with a hardware keyboard, like the iPad Pro with a Smart Keyboard. 

We can do this by implementing UIResponders' `keyCommands` method in the editor view controller and returning a list of `UIKeyCommand`s. Any commands that have a `discoverabilityTitle` will show up in the iOS overlay that appears when you hold down a modifier key on an external keyboard:

screen shot 2016-04-14 at 16 32 35

At the very least, we could add shortcuts that correspond to each of the items in the editor toolbar:

* Insert image (`Cmd+Shift+I`?)
* Bold and Italic we get for free (`Cmd+B`, `Cmd+I`), but I think we could add explicit keyboard shortcuts so that they show in the overlay
* Strikethrough (`Cmd+Shift+S`? `Cmd+~`?)
* Quote (`Cmd+>`?)
* Bullet list (OneNote uses `Cmd+.`, iA Writer uses `Cmd+L`)
* Ordered list (OneNote uses `Cmd+/`, iA Writer uses `Cmd+Shift+L`)
* Insert link (Cmd+K seems to be a fairly standard shortcut for this – it's what TextEdit uses)
* Toggle HTML mode (`Cmd+<`? `Cmd+H`?)

I can't find examples of pre-existing shortcuts for a number of these items in other apps (where I've used a ? above), so we'll just have to choose pick an appropriate shortcut in those cases.

### PR DESCRIPTION
**Pull request by frosty**
Fixes #806. I've added keyboard shortcuts for iPads using hardware keyboards. 

I've used shortcut keys they either: match system defaults (bold / underline), match other third party apps (bullet and ordered list shortcuts match iA Writer), or I've tried to pick something that makes sense where I couldn't find any precedent. Here's the initial selection that I've added:

keyboard-shortcuts

To test: Try to use each of the shortcuts listed above (you can hold down Command in the app to view the shortcuts). Test with text selected and without. 

Also, please let me know if you think any of the shortcut keys should be changed to something else!

Needs Review: @SergioEstevao 
Thanks!

### CODE REVIEW

**Comment by SergioEstevao** [comment]
@frosty Do you mind adding these shortcuts to the legacy editor also?

File: Classes/WPEditorViewController.m [hunk 1]
```diff
@@ -174,6 +174,55 @@ - (void)willTransitionToTraitCollection:(UITraitCollection *)newCollection withT
...
+    return YES;
+}
+
+- (NSArray<UIKeyCommand *> *)keyCommands
+{
+    return @[
+             [UIKeyCommand keyCommandWithInput:@"I"
+                                 modifierFlags:UIKeyModifierCommand|UIKeyModifierShift
+                                        action:@selector(didTouchMediaOptions)
+                          discoverabilityTitle:NSLocalizedString(@"Insert Image", @"Discoverability title for insert image keyboard shortcut.")],
```
Comment (SergioEstevao) [inline Classes/WPEditorViewController.m]: This shortcut should be renamed to be "Insert Media" and maybe change the shortcut to Cmd + M

**Comment by frosty** [comment]
@SergioEstevao I've managed to add support to the legacy editor. I had to add a `UITextView` subclass, and I've explained the reason for that in its documentation comments.

**Comment by SergioEstevao** [comment]
@frosty I tested it a bit more on the device and found the following issues:
 - on the visual editor, the shortcuts don't work  on the HTML mode
 - on the visual editor, when you use a shortcut, the corresponding icon on the toolbar should be activated. For bold, italic and underline the icons that are getting activated are the one on the new standard keyboard toolbar on the iPad, is this expected?

**Comment by frosty** [comment]
Thanks for the comments, @SergioEstevao – sorry for the delay in replying!

> on the visual editor, when you use a shortcut, the corresponding icon on the toolbar should be activated

This is working for me – can you confirm that it's definitely not for you? I can record a GIF / video if necessary.

> on the visual editor, the shortcuts don't work on the HTML mode

I've taken a look at this and it seems to be the same issue as with the legacy editor: HTML uses a `UITextView`, which suffers from the same iOS 9 bug. I'm just thinking about the best way to approach fixing this in both places (alongside working on other stuff), and I'll try and get something pushed up soon!

**Comment by frosty** [comment]
@SergioEstevao I've been looking into highlighting the toolbar buttons, and the problem only affects the Bold and Italic buttons. It's the same iOS 9 bug, where our custom methods for these shortcuts aren't called. The issue is that we seem to have no way of knowing then the user's pressed the keyboard shortcut. I've even hooked up the editor to Safari's inspector, and no javascript events get fired when you press Cmd+B/ Cmd+I. 

It's up to you whether you want to proceed without these two buttons getting highlighted as soon as you press the shortcut? They do highlight once you start pressing other keys, and as you noted the B / I icons on the top of the keyboard get highlighted too. I think the benefit of having the shortcuts in the first place outweighs these few issues.

**Comment by SergioEstevao** [comment]
Ok, let's proceed with this anyway I think it will be good for our users with keyboards, even if they are few of them. :shipit:

**Comment by frosty** [comment]
Thanks @SergioEstevao!

**Pull request merged**